alchemiscale_fah/compute/models.py

```diff
@@ -0,0 +1,149 @@
+from enum import Enum, auto
+from typing import Optional
+from ipaddress import IPv4Address
+from datetime import datetime
+from enum import Enum
+
+from pydantic import BaseModel, Field, validator
+
+from alchemiscale.models import ScopedKey
+from ..utils import NonbondedSettings
+
+
+# FahAdaptiveSamplingClient models
+
+
+class CompressionTypeEnum(Enum):
+    NONE = "NONE"
+    BZIP2 = "BZIP2"
+    ZLIB = "ZLIB"
+    GZIP = "GZIP"
+    LZ4 = "LZ4"
```

Comment:
Could use

```python
from enum import StrEnum, auto

class UpperStrEnum(StrEnum):
    @staticmethod
    def _generate_next_value_(name, *args):
        return name.upper()

# and then 

class CompressionTypeEnum(UpperStrEnum):
    NONE = auto()
    BZIP2 = auto()
```

OR could change to lowercase everywhere and use auto on `StrEnum` directly. 

Regardless should use `StrEnum` anyway as base, even if you don't use `auto` for following comparison to work properly 

```python
CompressionTypeEnum.NONE == "NONE" # true if `CompressionTypeEnum` is StrEnum, false if not.
```

Reply:
We'll switch to `StrEnum`; good catch on that one.

It took me a few mins to understand what the block above with `_generate_next_value_` does, and I'd rather keep the simplicity of what's there even if it looks a little repetitive. Doesn't feel like it saves us much.